### PR TITLE
*: support  cleanup audit logs  

### PR DESCRIPTION
**Pull request by srstack**
### What problem does this PR solve? 

support dm/cluster audit logs

### What is changed and how it works?
image

image

image

image

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Release notes:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cluster/audit/audit.go [hunk 1]
```diff
@@ -207,3 +208,104 @@ func decodeAuditID(auditID string) (time.Time, error) {
...
+		Size:  0,
+		Count: 0,
+	}
+
+	//  audit logs before `DelBeforeTime` will be deleted
+	oneDayDuration, _ := time.ParseDuration("-24h")
+	deleteLog.DelBeforeTime = time.Now().Add(oneDayDuration * time.Duration(retainDay))
+
+	fileInfos, err := os.ReadDir(dir)
+	if err != nil && !os.IsNotExist(err) {
```
Comment (nexustar) [inline pkg/cluster/audit/audit.go:236]: why && !os.IsNotExist(err)

File: pkg/cluster/audit/audit.go [hunk 2]
```diff
@@ -207,3 +208,104 @@ func decodeAuditID(auditID string) (time.Time, error) {
...
+	Files         []string  `json:"files"`
+	Size          int64     `json:"size"`
+	Count         int       `json:"count"`
+	DelBeforeTime time.Time `json:"delete_before_time"` // audit logs before `DelBeforeTime` will be deleted
+}
+
+// DeleteAuditLog  cleanup audit log
+func DeleteAuditLog(dir string, retainDay int, skipConfirm bool, displayMode string) error {
+	if retainDay < 0 {
+		return errors.Errorf("retain-time cannot be less than 0")
```
Comment (AstroProfundis) [inline pkg/cluster/audit/audit.go:222]: should be match the cli argument

Comment (srstack) [reply pkg/cluster/audit/audit.go]: It is to detect whether the input parameters are legal. Cli also needs to add this check.

**Review by AstroProfundis — APPROVED** [review]

**Pull request merged**